CRM/Contribute/WorkflowMessage/RecurringEdit.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+class CRM_Contribute_WorkflowMessage_RecurringEdit extends Civi\WorkflowMessage\GenericWorkflowMessage {
+  const GROUP = 'msg_tpl_workflow_contribution_recur';
```

Comment:
TBH, I'm not sure what functional purpose the group serves. It's clearly used for organizational purposes in the OG/OV setup, and there may be some correlation between the WF group and the range of accepted inputs/tokens (although I wouldn't bet anything on it practice or theory).

So mostly agree about being skeptical of them. OTOH, they do seem too be expected as part of `sendTemplate` and related hooks, and it's just a question of picking your battles.

Reply:
@totten - yeah we did some work on deprecating them - I *think* we can avoid them in the new code? -

Reply:
To my eye, the `GROUP` is not *internally* required on the send/render codepaths. Reading from `Civi\WorkflowMessage` => `renderTemplate` => `loadTemplate`, it appears to allow for nullability. Then it goes out via `CRM_Utils_Hook::alterMailContent()` to contrib. So that's the wildcard.

I think it's less about new *code* and more about new *workflows* or new *messages*. So:

* If one is up-converting an existing workflow message (eg `contribution_recurring_edit`), then one should include `GROUP` to provide continuity. (Whenever you want to break compat on that hook param, then you can omit/remove the `GROUP`.)
* If one is adding a new workflow message (eg `contribution_foo_bar`), then you can infer that nobody in contrib currently expects `contribution_foo_bar` to have a group. So in some sense you're not breaking things by omitting the group. (I don't really know who consumes `$params['groupName']`... or why... so I don't know if they expect `groupName` to strictly provided or contingently provided.)

Anywho, I think it's easier to include it for now (like you did) and treat removal as a distinct/separate revision in the contract.

Reply:
@totten last time round we actully logged issues on the extensions that were expecting group - so we kinda did that work in the past.